app/code/community/Bolt/Boltpay/Model/FeatureSwitch.php

```diff
@@ -50,12 +94,109 @@ public function updateFeatureSwitches()
         $switches = array();
         foreach ($switchesData as $switch) {
             $switches[$switch->name] = (object)array(
-                'value' => $switch->value,
-                'defaultValue' => $switch->defaultValue,
-                'rolloutPercentage' => $switch->rolloutPercentage
+                self::VAL_KEY => $switch->value,
+                self::DEFAULT_VAL_KEY => $switch->defaultValue,
+                self::ROLLOUT_KEY => $switch->rolloutPercentage
             );
         }
-        Mage::getModel('core/config')->saveConfig('payment/boltpay/featureSwitches', json_encode($switches));
+        $this->switches = $switches;
+        Mage::getModel('core/config')->saveConfig('payment/boltpay/featureSwitches', serialize($switches));
```

Comment:
Json coding doesn't work for this case, because I want associative array on the first level and object in second level.
If you think we can use serialize, I'll change code to have associative array on both level.
Please confirm @mccottry

Reply:
@vitaliyreznikov Since our plugin is going to be in the marketplace you definitely need to make whatever changes are necessary in order to ditch serialize.

Reply:
Got it. @mccottry What do you think it's better: use associative array on both levels or use objects on both levels?

Reply:
Done